backend/src/main/kotlin/app/ehrenamtskarte/backend/regions/database/repos/RegionsRepository.kt

```diff
@@ -33,4 +37,28 @@ object RegionsRepository {
     fun updateDataPolicy(region: RegionEntity, dataPrivacyText: String) {
         region.dataPrivacyPolicy = dataPrivacyText
     }
+
+    fun findRegionByPostalCode(postalCode: String, projectId: EntityID<Int>): RegionEntity {
+        try {
+            val csvInput = FileReader("src/main/resources/import/plz_ort_bayern.csv")
+            val records: Iterable<CSVRecord> = CSVFormat.RFC4180.parse(csvInput)
+            val postalCodes: MutableMap<String, String> = getPostalCodes(records)
+            val regionIdentifier = postalCodes[postalCode] ?: throw Exception("Region couldn't be found")
+            val regionId = RegionEntity.find { Regions.regionIdentifier eq regionIdentifier and (Regions.projectId eq projectId) }.single().id
+            return RegionEntity[regionId]
+        } catch (e: Exception) {
+            throw Exception("Couldn't read CSV")
+        }
+    }
```

Comment:
Reading the CSV file should be done only once when starting the backend. You can probably pass the `regionIdentifierByPostalCode` map into the webservice endpoint using the GraphQLContext.

Reply:
moved to the `common/webservice`